Add code coverage and refactor cli.

m-mujica opened the pull request.

Commit:
Add code coverage and refactor cli.

matthewp commented on .travis.yml:
```diff
@@ -19,3 +19,4 @@ notifications:
       - matthew@bitovi.com
     on_success: change
     on_failure: change
+script: npm run coverage:upload
```
This means the guide tests are not running in Travis.  We need to be able to run all of these things as well: https://github.com/donejs/donejs/blob/cli-refactor/package.json#L10

Commit:
Make sure guide tests run in Travis.

Commit:
Fix run-binary tests in windows.

The binary has a different name in windows, so I changed the test to
create the fake binary with the right name based on the OS it's running,
also this PR changes cmd-init to it re-uses the run-binary module which
has logic to handle the binary names based on the OS.

m-mujica commented:
@matthewp is there a way to rebuild that failing appveyor job? it seems to be a one time thing https://ci.appveyor.com/project/daffl/donejs/build/1.0.1246/job/gnfrlgaor43xgtam#L1897
